### PR TITLE
Managing assets

### PR DESCRIPTION
**Pull request by Kirweed**
-[ADD] Display assets modal
-[ADD] Display error when assets are set but related condition is not set
-[CHANGE] How the asset table and table header are displayed
-[ADD] Filtering assets by type and text insert by user
image
image

### CODE REVIEW

File: src/components/Buttons/ConditionButton/ConditionButton.tsx [hunk 1]
```diff
@@ -19,22 +20,47 @@ const ConditionButton = ({
   onClick,
 }: ConditionButtonProps) => {
   const [hover, setHover] = useState(false);
```
Comment (GabrLit) [inline src/components/Buttons/ConditionButton/ConditionButton.tsx]: should it be isHovered, setIsHovered?

File: src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx [hunk 2]
```diff
@@ -1,56 +1,12 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { CloseIcon, HistoryIcon } from "../../../assets/icons";
 import SearchInput from "../../Common/SearchInput/SearchInput";
```
Comment (GabrLit) [inline src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx:3]: Relative imports

File: src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx [hunk 3]
```diff
@@ -62,101 +18,130 @@ enum SearchAssetsModalOptions {
...
+interface SearchAssetsModalProps {
+  id: string;
+}
+
+const SearchAssetsModal = ({ id }: SearchAssetsModalProps) => {
   const [activeOption, setActiveOption] = useState(
     SearchAssetsModalOptions.ALL
   );
+  const [displayedAssets, setDisplayedAssets] = useState(accessibleAssets);
+  const [filteredByTypeAssets, setFilteredByTypeAssets] =
```
Comment (GabrLit) [inline src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx]: assetsFilteredByType would be more correct here I guess

File: src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx [hunk 4]
```diff
@@ -62,101 +18,130 @@ enum SearchAssetsModalOptions {
...
+  const filterAssets = () => {
+    if (activeOption === SearchAssetsModalOptions.ALL) {
+      setFilteredByTypeAssets(accessibleAssets);
+    } else {
+      setFilteredByTypeAssets(
+        accessibleAssets.filter(
+          (item) => item.market.toLowerCase() === activeOption.toLowerCase()
+        )
+      );
+    }
```
Comment (GabrLit) [inline src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx]:
```tsx
if (activeOption === SearchAssetsModalOptions.ALL) return setFilteredByTypeAssets(accessibleAssets); 

setFilteredByTypeAssets(
        accessibleAssets.filter(
          (item) => item.market.toLowerCase() === activeOption.toLowerCase()
        )
      );
    }
```

File: src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx [hunk 5]
```diff
@@ -62,101 +18,130 @@ enum SearchAssetsModalOptions {
...
-            <div className={styles.gridItem} style={{ textAlign: "end" }}>
-              {item.market}
-            </div>
-          </>
+      <SearchInput
+        placeholder="Search for assets"
+        value={searchValue}
+        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+          setSearchValue(e.target.value)
+        }
```
Comment (GabrLit) [inline src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx:95]: can destructure { target } directly and type it
https://stackoverflow.com/a/48443771

File: src/store/conditionsSlice.ts [hunk 6]
```diff
@@ -42,8 +45,12 @@ const conditionsSlice = createSlice({
         state.push(action.payload);
       }
     },
+    setIsAssetSet(state, action: PayloadAction<string>) {
+      const index = parseInt(action.payload.split("-")[1]);
+      state[index].isAssetSet = true;
```
Comment (GabrLit) [inline src/store/conditionsSlice.ts]: what if index is NaN? ParseInt can return NaN

**Review by GabrLit — APPROVED** [review]
Looks okay, some big logic blocks in jsx can be moved to separate components and files

File: src/components/AssetsBar/AssetsBar.tsx [hunk 7]
```diff
@@ -1,30 +1,53 @@
...
+  useAssetsSelector,
+} from "store/hooks";
+import SearchAssetsModal from "components/Modals/SearchAssetsModal/SearchAssetsModal";
+import styles from "./AssetsBar.module.scss";
 
 interface AssetsBarProps {
-  acronym?: string;
-  asset?: string;
-  market?: string;
   width?: string;
```
Comment (bmatlak03) [inline src/components/AssetsBar/AssetsBar.tsx:12]: should it be optional arg?

File: src/components/Buttons/ConditionButton/ConditionButton.tsx [hunk 8]
```diff
@@ -19,22 +20,47 @@ const ConditionButton = ({
...
+  const [isErrorHover, setErrorHover] = useState(false);
   const modal = useModalsSelector(id);
```
Comment (bmatlak03) [inline src/components/Buttons/ConditionButton/ConditionButton.tsx:24]: like above, should it be isErrorHovered, setIsErrorHovered?

File: src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx [hunk 9]
```diff
@@ -1,56 +1,12 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
```
Comment (bmatlak03) [inline src/components/Modals/SearchAssetsModal/SearchAssetsModal.tsx:1]: unneccessary `React` import?

File: src/store/conditionsSlice.ts (hunk 6 above)
Comment (bmatlak03) [reply src/store/conditionsSlice.ts]: and I guess that `parseInt()` can be replaced just with `Number(action.payload.split("-")[1])` or simply `+action.payload.split("-")[1]`

**Pull request merged**